### PR TITLE
Update tokens to 5.0.0

### PR DESCRIPTION
**Pull request by QuncCccccc**
This PR is to
* Update the Material Design tokens to 5.0.0
* 4 color roles in **Light mode** have different default values
   `onPrimaryContainer`: Primary10  -> Primary30 
   `onSecondaryContainer`: Secondary10 -> Secondary30
   `onTertiaryContainer`: Tertiary10 -> Tertiary30
   `onErrorContainer`: Error10 -> Error30
    
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;Screenshot 2024-08-13 at 12 53 03 PM

* new tokens added in list.json
* `md.comp.menu.list-item.*` tokens are deprecated and should be replaced by tokens in list.json and we've done the migration last year:)!(#122388)

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] I followed the [breaking change policy] and added [Data Driven Fixes] where supported.
- [x] All existing and new tests are passing.

### CODE REVIEW

File: dev/tools/gen_defaults/data/list.json [hunk 1]
```diff
@@ -57,7 +57,33 @@
...
+  "md.comp.list.list-item.selected.dragged.state-layer.color": "onSurface",
+  "md.comp.list.list-item.selected.dragged.state-layer.opacity": "md.sys.state.dragged.state-layer-opacity",
+  "md.comp.list.list-item.selected.dragged.trailing-icon.color": "onSurface",
+  "md.comp.list.list-item.selected.focus.label-text.color": "onSecondaryContainer",
+  "md.comp.list.list-item.selected.focus.leading-icon.color": "onSurface",
+  "md.comp.list.list-item.selected.focus.state-layer.color": "onSurface",
+  "md.comp.list.list-item.selected.focus.state-layer.opacity": "md.sys.state.focus.state-layer-opacity",
+  "md.comp.list.list-item.selected.focus.trailing-icon.color": "onSurface",
+  "md.comp.list.list-item.selected.hover.label-text.color": "onSecondaryContainer",
+  "md.comp.list.list-item.selected.hover.leading-icon.color": "onSurface",
```
Comment (TahaTesser) [inline dev/tools/gen_defaults/data/list.json:72]: Lots of new tokens here. Maybe we should have an issue tracking widget update for this.

**Review by TahaTesser — APPROVED** [review]
LGTM!

File: dev/tools/gen_defaults/data/list.json (hunk 1 above)
Comment (TahaTesser) [reply dev/tools/gen_defaults/data/list.json:72]: As i understand these tokens aren't currently used by any widgets that support these tokens. We can keep a issue tracker to make sure to use new tokens.

Comment (QuncCccccc) [reply dev/tools/gen_defaults/data/list.json:72]: Thanks for helping review this PR! Your are right, these tokens should be used for the menu item in selected status and looks like we don't have this feature yet:) I just created an issue so we can collect new tokens to make sure M3 defaults are up-to-date: https://github.com/flutter/flutter/issues/153397

**Review by guidezpl — APPROVED** [review]
Will there be an accompanying doc for this breaking change?

**Comment by QuncCccccc** [comment]
> Will there be an accompanying doc for this breaking change?

Should I send an email to [flutter-announce@](https://groups.google.com/g/flutter-announce)? It seems there's no failures in `customer_testing` and `Google testing`, so probably not counted as breaking changes, but we do have some golden test failures:) Looks like based on this [doc](https://github.com/flutter/flutter/blob/master/docs/contributing/Tree-hygiene.md#1-determine-if-your-change-is-a-breaking-change), we should just send an announcement. Any suggestions:)?

**Pull request merged**